### PR TITLE
feat: UI multi contracts per pipeline

### PR DESCRIPTION
**Pull request by lordmallam**
This implements the Mockup designs referenced in  [AET-348](https://jira.ehealthafrica.org/browse/AET-348) , [AET-349](https://jira.ehealthafrica.org/browse/AET-349)
Allowing a user to create multiple contracts displayed as tabs within a  pipeline with individual mapping validations based on the same input.

Also gets webpack working again with live and hot reloads

NOTE: 
Tasks outstanding:
- Contract revalidation on input change

@sabeen some styling updates are needed, especially around `read-only` pipelines and contracts

Related JIRA Ticket:
[AET-352](https://jira.ehealthafrica.org/browse/AET-352)

### CODE REVIEW

**Comment by obdulia-losantos** [comment]
Could you remove the `VERSION` and `REVISION` files? :grimacing:

File: aether-ui/aether/ui/assets/apps/pipeline/Pipeline.jsx [hunk 1]
```diff
@@ -190,16 +368,139 @@ class Pipeline extends Component {
...
+                id='settings.puplish.title'
+                defaultMessage='Publish status'
+              />
+            </label>
+            <div className='status-publish'>
+              <FormattedMessage
+                id='pipeline.publish-status'
+                defaultMessage={this.props.selectedContract.published_on
+                  ? `Published on ${moment(this.props.selectedContract.published_on).format('MMMM DD, YYYY HH:mm')}`
+                  : 'Not published'}
```
Comment (obdulia-losantos) [inline aether-ui/aether/ui/assets/apps/pipeline/Pipeline.jsx]: This is not the proper way of doing this. There should be an specific message for each id and not a dynamic one.

``` javascript
{ this.props.selectedContract.published_on &&
  <React.Fragment>
    <FormattedMessage
      id='pipeline.publish-status.yes'
      defaultMessage='Published on'
    /> { moment(this.props.selectedContract.published_on).format('MMMM DD, YYYY HH:mm') }
  </React.Fragment>
}
{ !this.props.selectedContract.published_on &&
  <FormattedMessage
    id='pipeline.publish-status.not'
    defaultMessage='Not published'
  />
}
```

File: aether-ui/aether/ui/assets/apps/pipeline/PipelineList.jsx [hunk 2]
```diff
@@ -114,66 +123,120 @@ class PipelineList extends Component {
         } header={this.state.errorHeader} show={this.state.showError}>
           {this.state.errorMessage}
         </Modal>
-        <div
-          onClick={() => { this.onSelectPipeline(pipeline) }}>
-          { pipeline.is_read_only &&
+        <div className={`preview-input ${pipeline.isInputReadOnly ? 'pipeline-readonly' : ''}`} onClick={this.onSelectPipeline.bind(this, pipeline)}>
+
```
Comment (obdulia-losantos) [inline aether-ui/aether/ui/assets/apps/pipeline/PipelineList.jsx]: Note: search for a javascript linter that complains with looooooooooooooooooooooooooog lines :unamused:

File: aether-ui/aether/ui/assets/apps/pipeline/PipelineList.jsx [hunk 3]
```diff
@@ -114,66 +123,120 @@ class PipelineList extends Component {
...
+                    </div>
+                  </div>
+
+                  <div className='contract-publish'>
+                    <div className='status-publish'>
+                      { pipeline.mappingset && <InfoButton pipeline={pipeline} /> }
+                      <FormattedMessage
+                        id='contract.list.publish-status'
+                        defaultMessage={contract.published_on ? `Published on ${moment(contract.published_on).format('MMMM DD, YYYY HH:mm')}`
+                          : 'Not published'}
```
Comment (obdulia-losantos) [inline aether-ui/aether/ui/assets/apps/pipeline/PipelineList.jsx]: Same comment about messages...

File: docker-compose-base.yml [hunk 4]
```diff
@@ -366,6 +366,8 @@ services:
       - ./aether-ui/aether/ui/assets/css:/code/css
       - ./aether-ui/aether/ui/assets/tests:/code/tests
       - ./aether-ui/aether/ui/assets/package.json:/code/package.json
+    environment:
+      WEBPACK_STATS_FILE: /code/aether/ui/assets/bundles/webpack-stats.json
```
Comment (obdulia-losantos) [inline docker-compose-base.yml]: This is useless.

**Review by obdulia-losantos — CHANGES_REQUESTED** [review]
I still need to test it locally, but apart from the "styling" issues only the messages comments need to be fixed.

**Comment by obdulia-losantos** [comment]
Shouldn't this be **create passthrough CONTRACT**?

image

**Comment by admbtlr** [comment]
Can you take another look @obdulia-losantos @sabeen ?

File: aether-ui/aether/ui/assets/apps/pipeline/Pipeline.jsx (hunk 1 above)
Comment (obdulia-losantos) [reply aether-ui/aether/ui/assets/apps/pipeline/Pipeline.jsx]: Thanks!

File: aether-ui/aether/ui/assets/apps/pipeline/PipelineList.jsx (hunk 3 above)
Comment (obdulia-losantos) [reply aether-ui/aether/ui/assets/apps/pipeline/PipelineList.jsx]: ~Thanks again!~ Oops! No, it wasn't fixed :disappointed:

**Review by obdulia-losantos — DISMISSED** [review]

**Review by sabeen — DISMISSED** [review]

**Review by obdulia-losantos — DISMISSED** [review]

File: aether-ui/aether/ui/assets/apps/pipeline/InfoButton.jsx [hunk 5]
```diff
@@ -1,7 +1,6 @@
 import React, { Component } from 'react'
```
Comment (obdulia-losantos) [inline aether-ui/aether/ui/assets/apps/pipeline/InfoButton.jsx:21]: Where is the LICENSE header?

Comment (admbtlr) [reply aether-ui/aether/ui/assets/apps/pipeline/InfoButton.jsx:21]: 😱

**Review by obdulia-losantos — APPROVED** [review]

**Pull request merged**